### PR TITLE
unaware_nodes whitelisted check

### PR DESCRIPTION
**Pull request by kruftik**
The proposed change allows to specify a whitelist of node flavours the rack configuration should be specified on. E.g. only data-nodes can be monitored for rack assignment.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: yt/odin/checks/bin/unaware_nodes/__main__.py [hunk 1]
```diff
@@ -15,7 +25,7 @@ def run_check(yt_client, logger, options, states):
     if allow_unaware_nodes:
         return states.FULLY_AVAILABLE_STATE
 
-    unaware_nodes = find_unaware_nodes(yt_client)
+    unaware_nodes = find_unaware_nodes(yt_client, ["data"])
```
Comment (gritukan) [inline yt/odin/checks/bin/unaware_nodes/__main__.py:28]: Let's make this list configurable with default being all possible flavors.

File: yt/odin/checks/bin/unaware_nodes/__main__.py [hunk 2]
```diff
@@ -1,9 +1,26 @@
 from yt_odin_checks.lib.check_runner import main
 
+DEFAULT_ENABLED_NODE_FLAVORS = [
+    "data",
+    "exec",
+    "tablet",
+    "chaos"
```
Comment (Kontakter) [inline yt/odin/checks/bin/unaware_nodes/__main__.py:7]: Let add comma for symmetry here

File: yt/odin/checks/bin/unaware_nodes/__main__.py [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+]
 
-def find_unaware_nodes(yt_client):
+
+def find_unaware_nodes(yt_client, check_node_flavors):
```
Comment (Kontakter) [inline yt/odin/checks/bin/unaware_nodes/__main__.py:11]: `check_node_flavors` – sounds like a boolean, maybe call it `node_flavors_to_check`? (here and in options)

File: yt/odin/checks/bin/unaware_nodes/__main__.py [hunk 4]
```diff
@@ -1,9 +1,26 @@
...
-    for row in yt_client.list("//sys/cluster_nodes", attributes=["rack"], read_from="cache"):
+    for row in yt_client.list("//sys/cluster_nodes", attributes=["rack", "flavors"], read_from="cache"):
+        flavors = row.attributes.get("flavors", [])
+
+        check_node_rack = False
+        for node_flavor in flavors:
+            if node_flavor in check_node_flavors:
+                check_node_rack = True
+                break
+        if not check_node_rack and len(flavors) > 0:
```
Comment (Kontakter) [inline yt/odin/checks/bin/unaware_nodes/__main__.py:21]: `len(flavors) > 0` -> `flavors`, it is a pythonic way to check list for non-emptyness

File: yt/odin/checks/bin/unaware_nodes/__main__.py [hunk 5]
```diff
@@ -1,9 +1,26 @@
 from yt_odin_checks.lib.check_runner import main
 
+DEFAULT_ENABLED_NODE_FLAVORS = [
```
Comment (gritukan) [inline yt/odin/checks/bin/unaware_nodes/__main__.py:3]: Don't you want to make default list empty (or none) and not check flavors in this case? New node flavors may appear from time to time and adding them here is a little bit counterintuitive.

Comment (kruftik) [reply yt/odin/checks/bin/unaware_nodes/__main__.py:3]: yeah, it makes sense

File: yt/odin/checks/bin/unaware_nodes/__main__.py [hunk 6]
```diff
@@ -1,25 +1,19 @@
...
-                break
-        if not check_node_rack and len(flavors) > 0:
-            continue
+        if node_flavors_to_check:
+            check_node_rack = False
+            for flavor in node_flavors:
+                if flavor in node_flavors_to_check:
+                    check_node_rack = True
+                    break
+            if not check_node_rack and node_flavors:
```
Comment (gritukan) [inline yt/odin/checks/bin/unaware_nodes/__main__.py:15]: Why do we check non-emptyness of node_flavors here? Each node at least has a "cluster" flavor.

Comment (kruftik) [reply yt/odin/checks/bin/unaware_nodes/__main__.py:15]: I'd looked into a couple of clusters available and didn't see such a flavor. I DID see it in a `.h` file but it has been marked as a deprecated one.

**Review by gritukan — APPROVED** [review]
LGTM

**Comment by Kontakter** [comment]
Shipped!

**Pull request closed without merging**